### PR TITLE
Port Postgres database to new data model

### PR DESCRIPTION
**Pull request by danielg0**
Reimplements the postgres interface for the new data model (#77)

### CODE REVIEW

File: internal/database/postgres/postgres.go [hunk 1]
```diff
@@ -148,57 +153,23 @@ func updateLastSeen(host string, now time.Time, tx *sql.Tx) (err error) {
...
+func (conn postgresConn) AppendDataPoint(sample uplink.GPUStatSample) error {
 	now := time.Now()
-	err = insertStats(id, packet, now, tx)
-	if err != nil {
-		return errors.Join(err, tx.Rollback())
-	}
 
-	return tx.Commit()
+	_, err := conn.db.Exec(`INSERT INTO Stats
+		(Gpu, Recieved, MemoryUtilisation, GpuUtilisation, MemoryUsed,
```
Comment (YRMYJASKA) [inline internal/database/postgres/postgres.go]: can we spell received properly now hehe

**Review by YRMYJASKA — COMMENTED** [review]
lgtm, but just had a quick look through. fine to merge if it passes all related tests

File: internal/database/postgres/postgres.go (hunk 1 above)
Comment (danielg0) [reply internal/database/postgres/postgres.go]: Damn it :laughing:. I swear I passed GCSE english. Will fix in a bit

**Comment by danielg0** [comment]
> lgtm, but just had a quick look through. fine to merge if it passes all related tests

The tests need to be modified in a few places, and the in-memory implementation still needs updating if we want to maintain that. I don't think it'll be too difficult to do

**Pull request merged**